packages/core/src/data/resolvers/queries/segments.ts

```
@@ -92,7 +92,15 @@ const segmentQueries = {
       const depServiceMeta = depService.config.meta || {};
 
       if (depServiceMeta.segments) {
-        const contentTypes = depServiceMeta.segments.contentTypes || [];
+        let contentTypes = depServiceMeta.segments.contentTypes || [];
+
+        if (!!dService?.types?.length) {
```

Biome (1.9.4)

[error] Avoid redundant double-negation.

It is not necessary to use double-negation when a value will already be coerced to a boolean.
Unsafe fix: Remove redundant double-negation


(lint/complexity/noExtraBooleanCast)